storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/RemoteLogMetadataManagerTestUtils.java

```diff
@@ -0,0 +1,119 @@
+
+        private Builder() {
+        }
+
+        public Builder bootstrapServers(String bootstrapServers) {
+            this.bootstrapServers = Objects.requireNonNull(bootstrapServers);
+            return this;
+        }
+
+        public Builder topicIdPartitions(Set<TopicIdPartition> topicIdPartitions) {
+            this.topicIdPartitions = Objects.requireNonNull(topicIdPartitions);
+            return this;
+        }
+
+        public Builder startConsumerThread(boolean startConsumerThread) {
+            this.startConsumerThread = startConsumerThread;
+            return this;
+        }
+
+        public Builder remoteLogMetadataTopicPartitioner(RemoteLogMetadataTopicPartitioner remoteLogMetadataTopicPartitioner) {
+            this.remoteLogMetadataTopicPartitioner = Objects.requireNonNull(remoteLogMetadataTopicPartitioner);
+            return this;
+        }
+
+        public Builder overrideRemoteLogMetadataManagerProps(Map<String, Object> overrideRemoteLogMetadataManagerProps) {
+            this.overrideRemoteLogMetadataManagerProps = Objects.requireNonNull(overrideRemoteLogMetadataManagerProps);
+            return this;
+        }
+
+        public TopicBasedRemoteLogMetadataManager build() {
+            String logDir = TestUtils.tempDirectory("rlmm_segs_").getAbsolutePath();
+            TopicBasedRemoteLogMetadataManager topicBasedRemoteLogMetadataManager = new TopicBasedRemoteLogMetadataManager(startConsumerThread);
+
+            // Initialize TopicBasedRemoteLogMetadataManager.
+            Map<String, Object> configs = new HashMap<>();
+            configs.put(REMOTE_LOG_METADATA_COMMON_CLIENT_PREFIX + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+            configs.put(BROKER_ID, 0);
+            configs.put(LOG_DIR, logDir);
+            configs.put(REMOTE_LOG_METADATA_TOPIC_PARTITIONS_PROP, METADATA_TOPIC_PARTITIONS_COUNT);
+            configs.put(REMOTE_LOG_METADATA_TOPIC_REPLICATION_FACTOR_PROP, METADATA_TOPIC_REPLICATION_FACTOR);
+            configs.put(REMOTE_LOG_METADATA_TOPIC_RETENTION_MS_PROP, METADATA_TOPIC_RETENTION_MS);
+
+            log.debug("TopicBasedRemoteLogMetadataManager configs before adding overridden properties: {}", configs);
+            // Add override properties.
+            configs.putAll(overrideRemoteLogMetadataManagerProps);
+            log.debug("TopicBasedRemoteLogMetadataManager configs after adding overridden properties: {}", configs);
+
+            topicBasedRemoteLogMetadataManager.configure(configs);
+            if (remoteLogMetadataTopicPartitioner != null) {
+                topicBasedRemoteLogMetadataManager.setRlmTopicPartitioner(remoteLogMetadataTopicPartitioner);
+            }
+            Assertions.assertDoesNotThrow(() -> waitUntilInitialized(topicBasedRemoteLogMetadataManager, 60_000));
+
+            topicBasedRemoteLogMetadataManager.onPartitionLeadershipChanges(topicIdPartitions, Collections.emptySet());
```

Comment:
this line can also be removed:

```
topicBasedRemoteLogMetadataManager.onPartitionLeadershipChanges(topicIdPartitions, Collections.emptySet());
```

Reply:
I add it back since there are tests need this function to initialize it

https://github.com/apache/kafka/blob/ee16eee5debe653b005afc5a3f4e0bf5486b09b0/storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/RemoteLogSegmentLifecycleTest.java#L315